src/lib.rs

```diff
@@ -1,37 +1,213 @@
+//!     $ cargo new my-project
+//!     $ cargo new xtask
+//!     ```
+//!
+//! * Open the workspace's Cargo.toml and add the following:
+//!     ```toml
+//!     [workspace]
+//!     members = [
+//!         "my-project",
+//!         "xtask",
+//!     ]
+//!     ```
+//!
+//! ## Add a command alias
+//!
+//! Create a `.cargo/config.toml` file and add the following content:
+//!
+//! ```toml
+//! [alias]
+//! xtask = "run --package xtask --"
+//! ```
+//!
+//! Now you can run your xtask package using:
+//!
+//! ```console
+//! cargo xtask
+//! ```
+//!
+//! ## Directory layout example
+//!
+//! If the name of the project package is `my-project`, the directory layout should
+//! look like this:
+//!
+//! ```console
+//! project
+//! ├── .cargo
+//! │   └── config.toml
+//! ├── Cargo.toml
+//! ├── my-project
+//! │   ├── Cargo.toml
+//! │   └── src
+//! │       └── ...
+//! └── xtask
+//!     ├── Cargo.toml
+//!     └── src
+//!         └── main.rs
+//! ```
+//!
+//! You can find more informations about xtask
+//! [here](https://github.com/cargo-xtask/).
+//!
+//! ## Use xtask-wasm as a dependency
+//!
+//! Finally, add the following to the xtask package's `Cargo.toml`:
+//!
+//! ```toml
+//! [dependencies]
+//! xtask-wasm = "0.1.0"
+//! ```
+//!
+//! # Usage
+//!
+//! This library gives you 3 [clap](https://docs.rs/clap/latest/clap/) structs:
+//!
+//! * [`Dist`] - Generate a distributed package for Wasm
+//! * [`Watch`] - Re-run a given command when changes are detected (using
+//!     [xtask-watch](https://github.com/rustminded/xtask-watch))
+//! * [`DevServer`] - Serve your project at a given IP address.
+//!
+//! They all implement [`clap::Parser`] allowing to add them easily to
+//! an existing CLI implementation and are flexible enough to be customized for
+//! most use-cases.
+//!
+//! You can find further information for each type at their documentation level.
+//!
+//! # Examples
+//!
+//! * A basic implementation could look like this:
+//!     ```rust,no_run
+//!     use std::process::Command;
+//!     use xtask_wasm::{anyhow::Result, clap};
+//!
+//!     #[derive(clap::Parser)]
+//!     enum Opt {
+//!         Dist(xtask_wasm::Dist),
+//!         Watch(xtask_wasm::Watch),
+//!         Serve(xtask_wasm::DevServer),
+//!     }
+//!
+//!
+//!     fn main() -> Result<()> {
+//!         let opt: Opt = clap::Parser::parse();
+//!
+//!         match opt {
+//!             Opt::Dist(dist) => {
+//!                 let dist = dist
+//!                     .dist_dir_path("dist")
+//!                     .static_dir_path("project/static")
+//!                     .app_name("project")
+//!                     .run_in_workspace(true)
+//!                     .run("project")?;
+//!
+//!                 log::info!("Built at {}", dist.dist_dir.display());
+//!
+//!             }
+//!             Opt::Watch(watch) => {
+//!                 let mut command = Command::new("cargo");
+//!                 command.args(["xtask", "dist"]);
```

Comment:
Should be cargo check because that's the default of cargo-watch.
```suggestion
//!                 command.args(["check"]);
```